gcloud.go

```diff
@@ -9,15 +9,42 @@ import (
 	"time"
 
 	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
 )
 
 const (
 	project = "cockroach-ephemeral"
 	domain  = "@cockroachlabs.com"
-	zone    = "us-east1-b"
 	vmUser  = "cockroach"
 )
 
+// ZoneList is a slice of strings that implements pflag's value
+// interface.
+type ZoneList []string
+
+var _ pflag.Value = &ZoneList{}
+
+// String returns a comma seperated list of zones. This is part
+// of pflag's value interface.
+func (zoneList ZoneList) String() string {
+	return strings.Join(zoneList, ",")
+}
+
+// Type returns the underlying type in string form. This is part of pflag's
+// value interface.
+func (*ZoneList) Type() string {
+	return "ZoneList"
+}
+
+// Set adds a new value to the ZoneList. It is the important part of
+// pflag's value interface.
+func (zoneList *ZoneList) Set(value string) error {
+	*zoneList = ZoneList(strings.Split(value, ","))
+	return nil
+}
+
+var zones = ZoneList{"us-east1-b", "us-west1-b", "europe-west2-b", "us-east1-b"}
```

Comment:
multi-zone definitely shouldn't be the default.

Reply:
I actually disagree pretty strongly. Global data is a very important story of the 2.0 release and we need to be running geo-distributed clusters as much as possible to catch issues. If someone _really_ needs a single dc cluster, then they can use the flag to opt out.

Reply:
Sure, but right now everyone's workflow it to do performance work that does not involve geo-distributed cluster. If you want to make it easy to have zones, use a `--geo` bool flag that automatically picks among the (flag default) zones. (eg: round-robin through the zones).
You don't be breaking anyone's existing use, and you still have a simple `--geo` invocation.

Reply:
While I agree with the importance of testing geo-distributed clusters, please don't move my cheese. More specifically, changing the default will break nightly usage of roachprod for perf tests, perhaps jepsen tests (not sure if it is used there yet), perhaps other stuff. I'm sure we have more important work to do then fixing up those problems and educating engineers about this change.

Reply:
We have more important work to do than increasing our testing of the headline theme of the 2.0 release? I'm sorry but I disagree. And educating engineers about the flag to turn it off (which at the moment is probably only the people doing single-dc performance work) seems like a pretty low investment.

I'm happy to fix up nightly perf tests (though arguably we should also be running the perf tests on a geo-distributed cluster as well). Same for Jepsen, though once the performance is good enough, seems like that should be tested geo-distributed.

That said, I think there does need to be a `--single-dc` option to disable it instead of the `--zones=us-east1-b,us-east1-b,us-east1-b,us-east1-b` you'd have to do with the PR as it currently stands.

Reply:
There are plenty of features and performance work being done that does **not** want geo-distributed clusters. We can switch to making it the default in the future, but especially when adding something which will most likely break on setup/GCE quotas/random other reasons, it's better to have it off by default.

You can make it simple by doing something along the lines of:
```
--zones=us-east1,us-west, europe
--geo=false
```

When `--geo` is false, all N nodes are created in the first zone specified in `--zones`.
When `--geo` is true, each zone gets `num_nodes / num_zones` nodes (either round-robin or 1-3 in zone 1, 4-6 in zone 2, 7-8 in zone 3).

This makes flipping the `--geo` flag the only thing necessary unless you want to also configure exactly which zones you want. Everyone currently using roachprod will keep using it as it. Anyone who wants geo-distributed clusters just sets `--geo`.

Reply:
> We have more important work to do than increasing our testing of the headline theme of the 2.0 release?

I might argue that some of the stability work on non-distributed clusters is as or more important. Regardless of the priority, I'm pretty sure you won't increase testing of geo-distributed clusters by forcing those clusters on users of `roachprod` by default. I'm massively in agreement that we should increase testing of geo-distributed clusters. And I fully support making it easy to create such clusters. But I'm confused about the insistence on changing the default for an existing tool. Seems a step too far in my opinion.

I'm going to stop arguing about this, though and leave Marc to carry the torch. I'll adjust if the default changes.

Reply:
I like your proposed flags a lot. And I don't want to hold up this PR on the default. Let's do your proposal, but do expect me to bring up revisiting the default soon. @m-schneider sgty?

Reply:
I'm happy revisiting. Even without switching the default we should look at which of our automated tests should set `--geo` pretty might right after the PR is merged.

Reply:
Not planning to get pulled into this discussion too much, but I basically agree with @danhhz (and will let him carry the 🔦) and wish we'd just make this the default and adjust the 2-3 places that don't want it.

> Regardless of the priority, I'm pretty sure you won't increase testing of geo-distributed clusters by forcing those clusters on users of roachprod by default. 

The long list of problems discovered by @dianasaur323 on the PM cluster suggests otherwise.

Reply:
> The long list of problems discovered by @dianasaur323 on the PM cluster suggests otherwise.

I wasn't suggesting there aren't problems with geo-distributed clusters, just that changing the default for `roachprod` won't increase testing of such clusters. What will increase the testing of geo-distributed clusters is making it trivially easy to create such clusters, which this PR accomplished both with and without a change to the default.

Reply:
> I wasn't suggesting there aren't problems with geo-distributed clusters, just that changing the default for roachprod won't increase testing of such clusters.

🤷‍♂️ I guess we're going to have to agree to disagree. More use = more testing and the most versatile testing we can get is everyone using geo-distributed clusters unless they know for a fact they care about the local commit latencies. If you find yourself wanting to create non-distributed clusters all the time, I wouldn't be opposed to sticking an env var in there that you can set in your `.bashrc` so that you get the default behavior you want. For anyone else who doesn't care, I can only see benefits to giving them a cluster that's at least somewhat indicative of the feature we want to have polished by 2.0. And that includes performance work too - I doubt there's an innate drive in many of us to get performance numbers of geo-distributed clusters, but that's something to look at as well.

Reply:
Can we please stop? We're not suggesting we **never** make geo-distributed setups the default, we're just saying not right away. May main reasons are:
* this is brand new behavior, keeping the existing one should be the default
* it's going to fail in fun and interesting ways, that's what cloud stuff does. Let's find and fix those first.

So I would propose the following sequence of events (without any fixed timeline, but "soon" comes to mind):
1. merge PR with default of `--geo=false`
1. find automated tests that use roachprod. For all of them, determine if they absolutely need single-region testing and set `--geo=false` for those. Set `--geo=true` progressively on others
1. get everyone to use the latest roachprod. Document and encourage use of geo-distributed testing.
1. flip the default to true and tell everyone.

Once the last is done and you still want to run in a single region, then you'll know about `--geo`.

If you want to make the timeline as short as possible, exercise `--geo=true` as much as you can, find issues, troubleshoot them, and fix them.